src/components/portal/portal.tsx

```diff
@@ -41,63 +41,79 @@ export interface EuiPortalProps {
   portalRef?: (ref: HTMLDivElement | null) => void;
 }
 
-export class EuiPortal extends Component<EuiPortalProps> {
+interface EuiPortalState {
+  portalNode: HTMLDivElement | null;
+}
+
+export class EuiPortal extends Component<EuiPortalProps, EuiPortalState> {
   static contextType = EuiNestedThemeContext;
   declare context: ContextType<typeof EuiNestedThemeContext>;
 
-  portalNode: HTMLDivElement | null = null;
-
   constructor(props: EuiPortalProps) {
     super(props);
+
+    this.state = {
+      portalNode: null,
+    };
+  }
+
+  componentDidMount() {
     if (typeof window === 'undefined') return; // Prevent SSR errors
 
     const { insert } = this.props;
 
-    this.portalNode = document.createElement('div');
-    this.portalNode.dataset.euiportal = 'true';
+    const portalNode = document.createElement('div');
+    portalNode.dataset.euiportal = 'true';
 
     if (insert == null) {
       // no insertion defined, append to body
-      document.body.appendChild(this.portalNode);
+      document.body.appendChild(portalNode);
     } else {
       // inserting before or after an element
       const { sibling, position } = insert;
-      sibling.insertAdjacentElement(insertPositions[position], this.portalNode);
+      sibling.insertAdjacentElement(insertPositions[position], portalNode);
     }
-  }
 
-  componentDidMount() {
-    this.setThemeColor();
-    this.updatePortalRef(this.portalNode);
+    this.setThemeColor(portalNode);
+    this.updatePortalRef(portalNode);
+
+    this.setState({
+      portalNode,
+    });
   }
 
   componentWillUnmount() {
-    if (this.portalNode?.parentNode) {
-      this.portalNode.parentNode.removeChild(this.portalNode);
+    const { portalNode } = this.state;
+    if (portalNode?.parentNode) {
+      portalNode.parentNode.removeChild(portalNode);
     }
     this.updatePortalRef(null);
   }
 
   // Set the inherited color of the portal based on the wrapping EuiThemeProvider
-  setThemeColor() {
-    if (this.portalNode && this.context) {
+  private setThemeColor(portalNode: HTMLDivElement) {
+    if (this.context) {
       const { hasDifferentColorFromGlobalTheme, colorClassName } = this.context;
 
       if (hasDifferentColorFromGlobalTheme && this.props.insert == null) {
-        this.portalNode.classList.add(colorClassName);
+        portalNode.classList.add(colorClassName);
       }
     }
   }
 
-  updatePortalRef(ref: HTMLDivElement | null) {
+  private updatePortalRef(ref: HTMLDivElement | null) {
     if (this.props.portalRef) {
       this.props.portalRef(ref);
     }
   }
 
   render() {
-    return this.portalNode ? (
-      <>{createPortal(this.props.children, this.portalNode)}</>
-    ) : null;
+    const { portalNode } = this.state;
+
+    if (!portalNode) {
+      return null;
+    }
+
+    return createPortal(this.props.children, portalNode);
```

Comment:
Just curious, was there a particular reason this couldn't be written as a ternary still? i.e.,

```suggestion
    const { portalNode } = this.state;

    return portalNode ? createPortal(this.props.children, portalNode) : null;
```

Reply:
Not really, but I find it easier to eye-scan when written like this. I like to leave ternary operators for inlining logic in places where you can't use regular if-else blocks, like function arguments or property values.

Reply:
Haha, I think we just have different approaches to ternaries then. I like using them for many things, they flow/read really clearly to me in my head, although I def appreciate that's not the case for everyone 😅 definitely not a blocker/change request, will collapse this thread